eras/conway/impl/src/Cardano/Ledger/Conway/Rules/VDel.hs

```diff
@@ -58,9 +75,34 @@ instance
   where
   type State (ConwayVDEL era) = VState era
   type Signal (ConwayVDEL era) = ConwayCommitteeCert (EraCrypto era)
-  type Environment (ConwayVDEL era) = VDelEnv era
+  type Environment (ConwayVDEL era) = PParams era
   type BaseM (ConwayVDEL era) = ShelleyBase
   type PredicateFailure (ConwayVDEL era) = ConwayVDelPredFailure era
   type Event (ConwayVDEL era) = ConwayVDelEvent era
 
-  transitionRules = undefined -- TODO
+  transitionRules = [conwayVDelTransition @era]
+
+conwayVDelTransition :: TransitionRule (ConwayVDEL era)
+conwayVDelTransition = do
+  TRC
+    ( _pp
+      , vState@VState {vsDReps, vsCommitteeHotKeys}
+      , c
+      ) <-
+    judgmentContext
+  case c of
+    ConwayRegDRep cred _deposit -> do
+      Set.notMember cred vsDReps ?! ConwayDRepAlreadyRegisteredVDEL cred
+      -- TODO: check against a new PParam `drepDeposit`, once PParams are updated. -- someCheck ?! ConwayDRepIncorrectDeposit deposit
+      pure $ vState {vsDReps = Set.insert cred vsDReps}
+    ConwayUnRegDRep cred _deposit -> do
+      -- TODO: check against a new PParam `drepDeposit`, once PParams are updated. -- someCheck ?! ConwayDRepIncorrectDeposit deposit
+      Set.member cred vsDReps ?! ConwayDRepNotRegisteredVDEL cred
+      pure $ vState {vsDReps = Set.delete cred vsDReps}
+    ConwayAuthCommitteeHotKey coldK hotK -> do
+      (isNothing <$> Map.lookup coldK vsCommitteeHotKeys) /= Just True ?! ConwayCommitteeHasResignedVDEL coldK
+      pure $ vState {vsCommitteeHotKeys = Map.insert coldK (Just hotK) vsCommitteeHotKeys}
+    ConwayResignCommitteeColdKey coldK -> do
+      Map.member coldK vsCommitteeHotKeys ?! ConwayCommitteeNotRegisteredVDEL coldK
```

Comment:
Oh okay, I will remove it now. This would allow a person who hasn't ever authorized a hot key, to simply resign. Thanks @WhatisRT :raised_hands:

Reply:
Yeah, it doesn't really matter since it doesn't change the state. But the logic in the spec is the same, if you've resigned you can't publish any certs for that cold key anymore. So that check makes sense to keep.

Reply:
Updated! :raised_hands: